reconcile/ocm/types.py

```diff
@@ -55,5 +55,9 @@ class OCMSpec(BaseModel):
 
     class Config:
         smart_union = True
-        # This is need to populate by either console_url or consoleUrl, for instance
-        allow_population_by_field_name = True
+        validate_assignment = True
+
+    @validator("server_url", "console_url", "elb_fqdn", pre=True, always=True)
```

Comment:
those fields are not optional. When we provision a cluster we set them as blank strings in the app-interface cluster spec.

Reply:
these fields are somehow just defined wrongly as mandatory in the gql schema because all integrations dealing with those fields need to be aware that they might be set to an empty string at least for a little while. but that is not something quickly fixed. i'm in favour of going forward with this PR in the meantime.